### PR TITLE
`ScotopiaPatch` fix and other corrections

### PR DESCRIPTION
**Pull request by IRacle1**
Don't blame me Nao😭

### CODE REVIEW

File: Exiled.API/Features/Roles/Scp106Role.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public class Scp106Role : Role
         /// <summary>
         /// Gets the <see cref="Scp106PlayerScript"/> script for the role.
         /// </summary>
-        public Scp106PlayerScript Script => script ??= Owner.ReferenceHub.scp106PlayerScript;
+        public Scp106PlayerScript Script => script ? script : script = Owner.ReferenceHub.scp106PlayerScript;
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Roles/Scp106Role.cs:31]: Why

File: Exiled.API/Features/Roles/Scp079Role.cs [hunk 2]
```diff
@@ -28,7 +28,7 @@ public class Scp079Role : Role
         /// <summary>
         /// Gets the <see cref="Scp079PlayerScript"/> script for the role.
         /// </summary>
-        public Scp079PlayerScript Script => script ??= Owner.ReferenceHub.scp079PlayerScript;
+        public Scp079PlayerScript Script => script ? script : script = Owner.ReferenceHub.scp079PlayerScript;
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Roles/Scp079Role.cs:31]: No

File: Exiled.API/Features/Roles/Scp0492Role.cs [hunk 3]
```diff
@@ -30,7 +30,7 @@ internal Scp0492Role(Player player)
         /// <summary>
         /// Gets the <see cref="Scp049_2PlayerScript"/> for this role.
         /// </summary>
-        public Scp049_2PlayerScript Script => script ??= Owner.ReferenceHub.characterClassManager.Scp0492;
+        public Scp049_2PlayerScript Script => script ? script : script = Owner.ReferenceHub.characterClassManager.Scp0492;
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Roles/Scp0492Role.cs:33]: No

File: Exiled.API/Features/Player.cs [hunk 4]
```diff
@@ -594,35 +594,35 @@ public bool NoClipEnabled
...
         /// Equivalent to checking the player's <see cref="Team"/>.
         /// </summary>
-        public bool IsScp => Role?.Team == Team.SCP;
+        public bool IsScp => Role?.Team is Team.SCP;
 
         /// <summary>
         /// Gets a value indicating whether or not the player's <see cref="RoleType"/> is any human rank.
         /// </summary>
-        public bool IsHuman => Role is not null && Role.Is(out HumanRole _);
+        public bool IsHuman => Role?.Team is not Team.SCP and not Team.RIP;
```
Comment (NaoUnderscore) [inline Exiled.API/Features/Player.cs:620]: No

**Review by NaoUnderscore — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Roles/Scp106Role.cs (hunk 1 above)
Comment (IRacle1) [reply Exiled.API/Features/Roles/Scp106Role.cs:31]: That check for unity objects, problem can be in caching objects that need to save not only in current scene(scp scripts dont need to save, but everything can be, if you sure that it won't lead to any errors - I revert it)

File: Exiled.API/Features/Player.cs (hunk 4 above)
Comment (IRacle1) [reply Exiled.API/Features/Player.cs:620]: Why, type-cast is faster that 2 get-methods?

Comment (NaoUnderscore) [reply Exiled.API/Features/Player.cs:620]: ` Gets a value indicating whether or not the player's <see cref="RoleType"/> is any human rank.`
Human rank is defined by `HumanRole`, and even if it's faster than two get methods, we can't just assume that NW won't add more non-human roles. "We can add a `and not Team.NewNonHumanTeam` if that's the case", yes, but it goes against the maintainability and it leads to more issues.

File: Exiled.API/Features/Roles/Scp106Role.cs (hunk 1 above)
Comment (NaoUnderscore) [reply Exiled.API/Features/Roles/Scp106Role.cs:31]: I don't really get it.
`script ??= Owner.ReferenceHub.scp106PlayerScript` is the same as `script ? script : script = Owner.ReferenceHub.scp106PlayerScript`.

Comment (IRacle1) [reply Exiled.API/Features/Roles/Scp106Role.cs:31]: No, check IL code

File: Exiled.API/Features/Player.cs (hunk 4 above)
Comment (IRacle1) [reply Exiled.API/Features/Player.cs:620]: Okay, i'll revert it

File: Exiled.API/Features/Roles/Scp106Role.cs (hunk 1 above)
Comment (IRacle1) [reply Exiled.API/Features/Roles/Scp106Role.cs:31]: Or ask Yamato, I already told him about diff

Comment (NaoUnderscore) [reply Exiled.API/Features/Roles/Scp106Role.cs:31]: Yeah whatever, it's the same thing in this case 🤷🏼‍♂️

**Pull request merged**